site/src/pages/docs/client-factory-builder.mdx

```diff
@@ -0,0 +1,125 @@
+                             // Increase the connection timeout to 5 seconds.
+                             .connectTimeoutMillis(5000)
+                             .build());
+final WebClient client = WebClient.builder("http://armeria.com")
+                                  .factory(factory)
+                                  .build();
+client.get("/api").aggregate().join();
+```
+
+The client created with the given <type://ClientFactory> will respect the settings such as connection timeout
+that you defined.
+You might notice that the clients, which are created using the same <type://ClientFactory>,
+share one connection pool. If you don't want to share the connections among the clients,
+you should use different <typeplural://ClientFactory>.
+For the complete list of properties, see <type://ClientFactoryBuilder>.
+
+<Tip>
+
+  You might want to take a look at <type://ClientBuilder> to distinguish the client-specific properties.
+
+</Tip>
+
+You can also use <type://ClientFactoryOptions> to build <type://ClientFactory>.
+It's more verbose than calling a dedicated setter method, but it can be useful when you need to
+set a series of options programmatically.
+
+## Lifecycle of `ClientFactory`
+
+Most clients for a <type://ClientFactory> need to send requests to another server for the lifetime of an application.
+Therefore, you will generally avoid closing the <type://ClientFactory>, except for perhaps in a shutdown hook,
+to avoid losing any connection to the other servers.
+For unit tests where a <type://ClientFactory> is only used for a single test or test class,
+it is appropriate to close it when done using it.
+
+## Specifying Netty `ChannelOption`s
+
+You can also specify Netty [`ChannelOption`](https://netty.io/4.1/api/io/netty/channel/ChannelOption.html)
+via <type://ClientFactoryBuilder#channelOption(ChannelOption,T)>:
+
+```java
+final ClientFactory factory =
+        ClientFactory.builder()
+                     // Set the size of send socket buffer as 4096 bytes
+                     .channelOption(ChannelOption.SO_SNDBUF, 4096)
+                     .build();
+WebClient.builder().factory(factory).build();
+```
+
+## Specifying different `EventLoop` threads for different endpoints
+
+By default, Armeria keeps at most 1 connection per endpoint (host and port pair) when using HTTP/2,
+which is desirable for typical HTTP/2 services. Armeria achieves this by assigning one `EventLoop` per endpoint
+from the <type://ServerConfig#workerGroup()>. However, there's a limit to the requests that
+a single connection(or `EventLoop` which is a thread) can handle, so you might want to increase
+the number of connections(or `EventLoop` threads) for certain or all endpoints.
+
+You can do this via <type://ClientFactoryBuilder#maxNumEventLoopsPerEndpoint(int)>:
+
+```java
+ClientFactory factory =
+         ClientFactory.builder()
+                      // A client will use maximum 5 EventLoops from the worker group per endpoint
```

Comment:
nit: This line looks a bit long to render in the code block. Could you check it locally?